crates/jit/src/instantiate.rs

```diff
@@ -45,11 +44,19 @@ pub enum SetupError {
     DebugInfo(#[from] anyhow::Error),
 }
 
+// A thread-safe wrapper for function pointers.
+struct FinishedFunctions(BoxedSlice<DefinedFuncIndex, *const [VMFunctionBody]>);
+
+// This is safe because the pointers are not modified and are only returned
+// through a reference bound to RawCompiledModule instance
+unsafe impl Send for FinishedFunctions {}
```

Comment:
Instead of unsafely asserting send/sync for `FinishedFunctions`, could a newtype wrapper around `*const [VMFunctionBody]` be made to narrow-the scope of send/sync? That way we're not unsafely asserting thread safety about `DefinedFuncIndex` or `BoxedSlice`